get-started/hallucination-agent/question-answer.md

```diff
@@ -0,0 +1,208 @@
+- **Context validation mode** - when context is provided, the agent only validates answers against the specified context.
+
+### Fact-checking mode
+
+This example checks whether an answer contains hallucinated information. As no context is provided, the answer will be fact-checked against general knowledge to identify hallucinations.
+
+??? example "Python"
+
+    ```python
+    import requests
+    from getpass import getpass
+
+    # Get API key from user input
+    api_key = getpass("Enter your kluster.ai API key: ")
+
+    # Set up request data
+    url = "https://api.kluster.ai/v1/judges/detect-hallucination"
+    headers = {
+        "Authorization": f"Bearer {api_key}",
+        "Content-Type": "application/json"
+    }
+    payload = {
+        "prompt": "Is earth flat?",
+        "output": "Yes, my friend",
+        "return_search_results": False #Optional
+    }
+
+    # Send the request to the hallucination detection endpoint
+    response = requests.post(url, headers=headers, json=payload)
+
+    # Convert the response to JSON
+    result = response.json()
+
+    # Extract key information
+    is_hallucination = result.get("is_hallucination")
+    explanation = result.get("explanation")
+
+    # Print full response
+    print(f"🔗 API Response: {result}")
+
+    # Print whether hallucination was detected
+    print(f"{'HALLUCINATION DETECTED' if is_hallucination else 'NO HALLUCINATION DETECTED'}")
+
+    # Print the explanation 
+    print(f"\nExplanation: {explanation}")
+    ```
+??? example "CLI"
+
+    ```bash
+    #!/bin/bash
+
+    # Check if API_KEY is set and not empty
+    if [[ -z "$API_KEY" ]]; then
+        echo -e "\nError: API_KEY environment variable is not set.\n" >&2
+    fi
+
+    # Submit hallucination detection request
+    curl --location 'https://api.kluster.ai/v1/judges/detect-hallucination' \
+    --header "Authorization: Bearer $API_KEY" \
+    --header "Content-Type: application/json" \
+    --data '{
+        "prompt": "Is earth flat?",
+        "output": "Yes, my friend",
+        "return_search_results": false 
+    }'
+    ```
+
+### Context validation mode
+
+When providing the `context` parameter, the agent will not perform external fact-checking. Instead, it focuses on whether the answer complies with the provided context.
+
+!!! tip "RAG applications"
+    Ensure the LLM's responses are accurate by using the Hallucination Detection Agent in your Retrieval Augmented Generation workflows.
+
+
+This example checks whether an answer is correct based on the provided context.
+
+??? example "Python"
+
+    ```python
+    import requests
+    from getpass import getpass
+
+    # Get API key from user input
+    api_key = getpass("Enter your kluster.ai API key: ")
+
+    # Set up request data
+    url = "https://api.kluster.ai/v1/judges/detect-hallucination"
+    headers = {
+        "Authorization": f"Bearer {api_key}",
+        "Content-Type": "application/json"
+    }
+    payload = {
```

Comment:
minor nitpick but try to use standardized snippet variable names etc. Check the quickstart snippets to use as reference, this way, readers get used to variable names and understand what is this here etc